tests/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/ProjectFileEditorTests.cs

```diff
@@ -0,0 +1,256 @@
+        {
+            var solution = VisualStudio.ObjectModel.Solution;
+
+            ProjectTestExtension consoleProject;
+
+            using (Scope.Enter("Create Project"))
+            {
+                consoleProject = solution.CreateProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreConsoleApp);
+            }
+
+            using (Scope.Enter("Verify Create Project"))
+            {
+                solution.Verify.HasProject();
+            }
+
+            using (Scope.Enter("Open project file editor"))
+            {
+                consoleProject.DoubleClick();
+            }
+
+            using (Scope.Enter("Change project property"))
+            {
+                VisualStudio.ObjectModel.Search.FindReplace.ReplaceAllInFiles("Exe", "Library", LookInFilesSearchScopes.EntireSolution);
+            }
+
+            using (Scope.Enter("Verify Text Buffer Changed"))
+            {
+                var editor = VisualStudio.ObjectModel.WindowManager.ActiveDocumentWindowAsTextEditor;
+                editor.Editor.Verify.ContainsText("<OutputType>Library</OutputType>");
+            }
+
+            using (Scope.Enter("Verify Project Unchanged"))
+            {
+                consoleProject.Properties.Verify.PropertyValueIs(ProjectProperty.OutputType, prjOutputType.prjOutputTypeExe);
+            }
+        }
+
+        [TestMethod]
+        public void ProjectFileEditor_ReplaceAllWhileEditorNotLoaded_ReplacementsLoaded()
+        {
+            var solution = VisualStudio.ObjectModel.Solution;
+
+            ProjectTestExtension consoleProject;
+
+            using (Scope.Enter("Create Project"))
+            {
+                consoleProject = solution.CreateProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreConsoleApp);
+            }
+
+            using (Scope.Enter("Verify Create Project"))
+            {
+                solution.Verify.HasProject();
+            }
+
+            using (Scope.Enter("Change project property"))
+            {
+                VisualStudio.ObjectModel.Search.FindReplace.ReplaceAllInFiles("Exe", "Library", LookInFilesSearchScopes.EntireSolution, keepModifiedFilesOpen: KeepModifiedFilesOpen.No).WaitForReplaceInFilesCompletion();
+            }
+
+            using (Scope.Enter("Open project file editor"))
+            {
+                consoleProject.DoubleClick();
+            }
+
+            using (Scope.Enter("Verify Text Buffer Changed"))
+            {
+                var editor = VisualStudio.ObjectModel.WindowManager.ActiveDocumentWindowAsTextEditor;
+                editor.Editor.Verify.ContainsText("<OutputType>Library</OutputType>");
+            }
+
+            using (Scope.Enter("Verify Project Unchanged"))
+            {
+                // project reload is on a one second delay
+                Services.Synchronization.WaitFor(TimeSpan.FromSeconds(1), () => consoleProject.Properties.Verify.PropertyValueIs(ProjectProperty.OutputType, prjOutputType.prjOutputTypeLibrary));
```

Comment:
Why wait this way rather than, say, `await Task.Delay(...)`?

Reply:
Mostly a guess that this was "idiomatic" apex, and partly because if the `DoubleClick()` call takes a while (and apex doesn't seem speedy), this might not have to wait at all.

Reply:
Oh, this polls for the condition?

Reply:
Yeah, the first parameter is a timeout but it will exit early if the condition passes.